src/service/packet_router.proto

```diff
@@ -52,10 +52,38 @@ message packet_router_register_v1 {
   bytes signature = 3;
 }
 
+// Session offer from the packet router to the gateway. If the gateway wants to
+// initiate a session key it should send a packet_router_session_init_v1 message
+// to the packet router using data in this offer to propose a session key.
+//
+// This message is sent by packet router after the register command is sent by
+// the gateway.
+message packet_router_session_offer_v1 { bytes data = 1; }
+
+// Initializes a session key with the packet router. The data in this init
+// message has to be from the latest session offer the gateway has received, the
+// session key is the public key that is to be used to verify packets by packet
+// router, and this message is to be signed by the private key of the gateway.
+// Once accepted the session key replaces the current (default is gateway) key
```

Comment:
I think bytes `gateway = 9;` in `packet_router_packet_up_v1` above should signal that it is not always the gateway but the public session key also, maybe we can come up with a middle ground?

Reply:
I don't agree here.. The session key is a protocol optimization between gateways and HPR.. By keeping the gateway as is you're only overriding the signature if an alternate verifier was set as part of session negotiation

Reply:
OK, I see. So HPR should now about session (obviously) and check signature based on that knowledge only?

Reply:
yeah.. basically the "verifier" used in the packet router is set to the gateway public key as part of registration (or before?) and the session key becomes the verifier.. you could technically ignore the gateway bytes in the packet report altogether in that model